syntax = "proto3";

package user;

// 用户域的聚合
message UserDTO {
    // 聚合根 
    uint64 userID = 1;
    // 将不同业务需要的DTO分开定义，这样某个业务需要哪个DTO就序列化对应的，不需要的就空着
    // 如果各个DTO里面的字段都拿出来放在一起定义，有些业务只需要其中一部分字段，却还得序列化其他没用的字段，造成网咯数据传输的带宽浪费
    DeviceDTO device = 2;
    SettingDTO setting = 3;   // 这里把值对象也单独拿出来定义
    InformationDTO information = 4;
    ProfileDTO  pprofile = 5;
}

message DeviceDTO {
    uint64 deviceID = 1;
    string os = 2;
    string appVersion = 3;
    string type = 4;
    string model = 5;
}

message SettingDTO {
    string fontSize  = 1;
    bool darkMode = 2;
    bool receiveNotification = 3;
    string language = 4;
    bool notifications = 5;
 }

 // 个人资料信息，在个人信息页面展示
message InformationDTO {
    // 昵称
    string nickname = 1;
    // 密码
    string password = 2;
    // 头像
    string avatar = 3;
    // 个性签名
    string signature = 4;
}

// 异步任务构建用户画像
// 其实这就是个人详细资料
message ProfileDTO {
    string location = 1;
    int32 age = 2;
    string gender = 3;
    string tags = 4;
}

